server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -70,7 +71,7 @@ public static List<String> splitSmart(String s, String separator, boolean decode
             if (s.startsWith(separator, pos)) {
                 if (sb.length() > 0) {
                     lst.add(sb.toString());
-                    sb = new StringBuilder();
+                    sb.setLength(0);
```

Comment:
Turns out this method is only really used in one place...

Reply:
I see that this method is used by our command line parsing, so I think it shouldn't be marked as deprecated and thanks for adding a test for it. Even one usage is enough and it might be more used in the future.

Reply:
It was being used in one specific way, the other functionality (`decode` true) isn't used at all. I've refactored the use in EmbeddedCli away from this method, so now it isn't used at all.